contracts/WakuRlnRegistry.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {WakuRln} from "./WakuRln.sol";
+import {IPoseidonHasher} from "rln-contract/PoseidonHasher.sol";
+import {Ownable} from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import "forge-std/console.sol";
+
+error StorageAlreadyExists(address storageAddress);
+error NoStorageContractAvailable();
+error IncompatibleStorage();
+error IncompatibleStorageIndex();
+
+contract WakuRlnRegistry is Ownable {
+    uint16 public nextStorageIndex;
+    mapping(uint16 => address) public storages;
+
+    uint16 public usingStorageIndex = 0;
+
+    IPoseidonHasher public immutable poseidonHasher;
+
+    event NewStorageContract(uint16 index, address storageAddress);
+
+    constructor(address _poseidonHasher) Ownable() {
+        poseidonHasher = IPoseidonHasher(_poseidonHasher);
+    }
+
+    function _insertIntoStorageMap(address storageAddress) internal {
+        storages[nextStorageIndex] = storageAddress;
+        emit NewStorageContract(nextStorageIndex, storageAddress);
+        nextStorageIndex += 1;
```

Comment:
```suggestion
        unchecked {
            nextStorageIndex += 1;
        }
```

^ Take this with a huge grain of salt. Don't actually know how unlikely this is to overflow

Reply:
I should probably add a check before the storage is created to see if it overflows. we can then make this unchecked

Reply:
Hm.. so in case it does overflow it will revert by default. 
So if you want to add a check before incrementing, we might as well not add the `unchecked` block in the first place.
That should lead to the same result.

Reply:
Sounds good to me!